src/InputGroup/InputGroup.js

```diff
@@ -1,272 +1,56 @@
-                                    {glyph ? (
-                                        <Icon
-                                            disableStyles={disableStyles}
-                                            glyph={glyph}
-                                            role='presentation' />
-                                    ) : (
-                                        addon
-                                    )}
-                                </span>
-                            )}
-                            <FormInput
-                                {...inputProps}
-                                className={inputClasses}
-                                compact={compact}
-                                disableStyles={disableStyles}
-                                id={inputId}
-                                name={inputName}
-                                onChange={this.handleTextChange}
-                                value={this.state.value} />
-                        </div>
-                    );
-                } else {
-                    return (
-                        <div
-                            {...props}
-                            className={inputGroupClasses}>
-                            <FormInput
-                                {...inputProps}
-                                className={inputClasses}
-                                compact={compact}
-                                disableStyles={disableStyles}
-                                id={inputId}
-                                name={inputName}
-                                onChange={this.handleTextChange}
-                                value={this.state.value} />
-                            {actions ? (
-                                <span className={addonClasses}>
-                                    {children}
-                                </span>
-                            ) : (
-                                <span className={addonClasses}>
-                                    {glyph ? (
-                                        <Icon
-                                            disableStyles={disableStyles}
-                                            glyph={glyph}
-                                            role='presentation' />
-                                    ) : (
-                                        addon
-                                    )}
-                                </span>
-                            )}
-                        </div>
-                    );
-                }
-            }
-        }
     }
 }
 
+InputGroup.Addon = InputGroupAddon;
+
 InputGroup.displayName = 'InputGroup';
 
 InputGroup.propTypes = {
-    actions: PropTypes.bool,
-    addon: PropTypes.string,
-    addonClassNames: PropTypes.string,
-    addonPos: PropTypes.oneOf(INPUT_GROUP_ADDON_POSITIONS),
     children: PropTypes.node,
     className: PropTypes.string,
     compact: PropTypes.bool,
```

Comment:
We have a choice to make regarding `compact` and I think there are merits for each approach.  If we take a `compact` prop here, it is simply to pass along to its children (and subsequently through `InputGroupAddon` to its children) the prop so the proper classes get added to the proper elements.  The other option would be to _not_ take the prop here and just have the consumer pass their `compact` prop to their respective inputs and buttons manually.

Reply:
I've kept the prop for now to get it simple for users based on team feedback, but can be up for further debate.